packages/@azure/eventhubs/processor/lib/hostContext.ts

```diff
@@ -242,21 +242,30 @@ export namespace HostContext {
     };
     ctxt.getHubRuntimeInformation = async () => {
       const client = ctxt.getEventHubClient();
-      const result = await client.getHubRuntimeInformation();
-      client.close().catch(/* do nothing */);
-      return result;
+      try {
+        const result = await client.getHubRuntimeInformation();
+        return result;
+      } finally {
+        client.close().catch(/* do nothing */);
+      }
```

Comment:
Doesnt it make sense to wait until the client is closed before returning from these functions?

Reply:
That would change the existing behavior? Looks like there functions are for getting information and don't care about errors in closing so maybe little value in waiting?

Reply:
> Looks like there functions are for getting information and don't care about errors in closing so maybe little value in waiting

The value in waiting is not as much as to care about errors, but to ensure that we clean up the resources that were created just for this function.

If I am creating a client just for the scope of my function, then I should wait until it is closed before returning from the function.

Either that, or create the client once and use it for the lifetime of the `ctxt` object which I believe we cant do because the `ctxt` object doesnt have a "close" concept :)

Can you point me to one of the test cases that you intend to improve with this change?

Reply:
I agreed on what you said above. But that is a different issue than what I am addressing here, which is when for example, `client.getHubInformation()` throws, `client.close().catch()` is not executed.

The old code intentionally ignores errors from `client.close()` with `.catch(/* do nothing */)`. We should re-visit the design of how to handle errors from `client.close()` in another issue.

The hanging tests are
https://github.com/Azure/azure-sdk-for-js/blob/94e6116b01184f12dd011499044cb0c56fbb2f30/packages/%40azure/eventhubs/processor/tests/eph.spec.ts#L464-L505